pkg/cloudprovider/openstack.go

```diff
@@ -554,7 +617,38 @@ func (o *OpenStack) reserveNeutronIPAddress(s neutronsubnets.Subnet, ip net.IP,
 		Name:        fmt.Sprintf("egressip-%s", ip.String()),
 	}
 	p, err := neutronports.Create(o.neutronClient, opts).Extract()
+
 	if err != nil {
+		// Let's check if error suggests that port with that IP already exists in that subnet.
+		if _, ok := err.(*gophercloud.ErrDefault409); ok {
+			// If so, let's get the port and check if it's our reservation port. It's possible we've created it earlier.
+			// It's also possible user configured a taken range, we should erorr out in such case.
+			page, nestedErr := neutronports.List(o.neutronClient, neutronports.ListOpts{
+				NetworkID:   s.NetworkID,
+				FixedIPs:    []neutronports.FixedIPOpts{{SubnetID: s.ID, IPAddress: ip.String()}},
+				DeviceOwner: egressIPTag,
+				DeviceID:    generateDeviceID(serverID),
+			}).AllPages()
+			if nestedErr != nil {
```

Comment:
BTW - what's the correct pattern to nest errors? This one smells bad. I guess a separate function might look better?

Reply:
you can wrap errors with %w:
https://rollbar.com/blog/golang-wrap-and-unwrap-error/#

There's a repo that implements this:
github.com/pkg/errors
With Wrap functions

You can join multiple errors with:
import (
apierrors "k8s.io/apimachinery/pkg/util/errors"
)
 apierrors.NewAggregate(errors)
https://pkg.go.dev/k8s.io/apimachinery/pkg/util/errors#NewAggregate

Reply:
I just changed the logic instead, I wasn't exactly wanting to wrap errors.